client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java

```diff
@@ -254,38 +258,42 @@ public ControllerImpl(ManagedChannelBuilder<?> channelBuilder) {
 
     @Override
     public CompletableFuture<Boolean> scaleStream(final Stream stream, final List<Integer> sealedSegments,
+                                                  final Map<Double, Double> newKeyRanges, final long timeoutMillis,
```

Comment:
Actually, given the split of scale stream into start scale and check status, we could leave to the application to keep checking the status and avoid the problem altogether. 

The one problem I see is the break of symmetry. Other calls do not have this "check status" behavior. I suppose that the current behavior looks more like a "long poll".